akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -346,9 +349,12 @@ final class Split[T](decision: Split.SplitDecision, p: T ⇒ Boolean) extends Gr
             closeThis(handler, elem)
             handOver(handler)
           } else {
-            // Drain into the void
-            if (substreamCancelled) pull(in)
-            else substreamSource.push(elem)
+            if (substreamCancelled) {
+              if (eagerClose) completeStage()
```

Comment:
WDYM? `Tcp.outgoingConnection` has a `halfClose` parameter, which means something different. “close” is not properly defined for a stage, we have _complete_, _fail_, _cancel_ signals.

We still have eagerClose in SslTlsStage (to be rewritten soon) and on Interleave (which should be changed to eagerComplete).